utils/dwarf.c

```diff
@@ -262,18 +272,18 @@ static int setup_dwarf_info(const char *filename, struct uftrace_dbg_info *dinfo
 	if (force || check_trace_functions(filename) != TRACE_CYGPROF)
 		dinfo->needs_args = true;
 
-	pr_dbg2("setup dwarf debug info for %s\n", filename);
-
 	fd = open(filename, O_RDONLY);
 	if (fd < 0) {
 		pr_dbg2("cannot open debug info for %s: %m\n", filename);
 		return -1;
 	}
 
-	dinfo->dw = dwarf_begin(fd, DWARF_C_READ);
+	dinfo->dwfl = dwfl_begin(&callbacks);
+	dinfo->dwfl_module = dwfl_report_elf(dinfo->dwfl, filename, filename, fd, 0, true);
+	dwfl_report_end(dinfo->dwfl, NULL, NULL);
```

Comment:
I was worrying about the potential version compatibility issues, but it seems the dwfl APIs were supported long ago.